build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -439,6 +439,115 @@ service ContentAddressableStorage {
+  // However, clients MUST first check the server capabilities, whether blob
+  // splitting is supported by the server.
+  //
+  // Clients SHOULD verify whether the digest of the blob assembled by the
+  // fetched chunks results in the requested blob digest.
+  //
+  // Since the generated chunks are stored as blobs, they underlie the same
+  // lifetimes as other blobs. In particular, the chunk lifetimes are
+  // independent from the lifetime of the original blob:
+  //  * A blob and any chunk derived from it may be evicted from the CAS at
+  //    different times.
+  //  * A call to Split extends the lifetime of the original blob, and sets
+  //    the lifetimes of the resulting chunks (or extends the lifetimes of
+  //    already-existing chunks).
+  //  * Touching a chunk extends its lifetime, but does not extend the
+  //    lifetime of the original blob.
+  //  * Touching the original blob extends its lifetime, but does not extend
+  //    the lifetimes of chunks derived from it.
+  //
+  // When blob splitting and splicing is used at the same time, the clients and
+  // the server SHOULD agree out-of-band upon a chunking algorithm used by both
+  // parties to benefit from each others chunk data and avoid unnecessary data
+  // duplication.
+  //
+  // Errors:
+  //
+  // * `NOT_FOUND`: The requested blob is not present in the CAS.
+  // * `RESOURCE_EXHAUSTED`: There is insufficient disk quota to store the blob
+  //   chunks.
+  rpc SplitBlob(SplitBlobRequest) returns (SplitBlobResponse) {
+    option (google.api.http) = { get: "/v2/{instance_name=**}/blobs/{blob_digest.hash}/{blob_digest.size_bytes}:splitBlob" };
+  }
+
+  // Splice a blob from chunks.
+  //
+  // This is the complementary operation to the
+  // [ContentAddressableStorage.SplitBlob][build.bazel.remote.execution.v2.ContentAddressableStorage.SplitBlob]
+  // function to handle the chunked upload of large blobs to save upload
+  // traffic.
+  //
+  // If a client needs to upload a large blob and is able to split a blob into
+  // chunks in such a way that reusable chunks are obtained, e.g., by means of
+  // content-defined chunking, it can first determine which parts of the blob
+  // are already available in the remote CAS and upload the missing chunks, and
+  // then use this API to instruct the server to splice the original blob from
+  // the remotely available blob chunks.
+  //
+  // Servers MAY implement this functionality, but MUST declare whether they
+  // support it or not by setting the
+  // [CacheCapabilities.blob_splice_support][build.bazel.remote.execution.v2.CacheCapabilities.blob_splice_support]
+  // field accordingly.
+  //
+  // Clients MAY use this functionality, it is just an optimization to reduce
+  // upload traffic, when uploading large blobs to the CAS. However, clients
+  // MUST first check the server capabilities, whether blob splicing is
+  // supported by the server.
```

Comment:
How about "Clients MUST check that the server supports this capability, before deciding to use it" ?